src/project-manager.ts

```diff
@@ -892,9 +891,14 @@ export class ProjectConfiguration {
 			return;
 		}
 
+		// paths have been forward-slashified by TS (see toUnixPath below)
+		// TODO: consider moving expectedFilePaths out of LanguageServiceHost?
+		const expectedFilePaths = this.getHost().expectedFilePaths;
+
 		for (const uri of this.fs.uris()) {
 			const fileName = uri2path(uri);
-			if (isGlobalTSFile(fileName) || (!isDependencyFile(fileName) && isDeclarationFile(fileName))) {
+			if (isGlobalTSFile(fileName) ||
+				(isDeclarationFile(fileName) && expectedFilePaths.includes(toUnixPath(fileName)))) {
```

Comment:
At first I though this reverted https://github.com/tomv564/javascript-typescript-langserver/commit/021d1b455c8ba48a0f471b96cccb10dd78c23ed5 but it still looks in `this.fs.uris()` 👍 

> - when adding basic files, search for them project-wide instead of searching in expected files list. The reason is that files may be excluded by tsconfig.json rule